riff-raff/test/restrictions/RestrictionCheckerTest.scala

```diff
@@ -0,0 +1,168 @@
+package restrictions
+
+import java.util.UUID
+
+import com.gu.googleauth.UserIdentity
+import controllers.ApiKey
+import deployment.{ApiRequestSource, ContinuousDeploymentRequestSource, UserRequestSource}
+import org.joda.time.DateTime
+import org.scalatest.{FlatSpec, Matchers}
+
+class RestrictionCheckerTest extends FlatSpec with Matchers {
```

Comment:
There are a lot of tests here for "does my version of an if statement work like an if statement"